codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/Instantiator.kt

```diff
@@ -422,8 +430,14 @@ open class Instantiator(
                 }
         }
 
-        data.members.forEach { (key, value) ->
-            val memberShape = shape.expectMember(key.value)
+        for ((key, value) in data.members) {
+            val memberShape =
+                shape.getMember(key.value).getOrNull()
+                    ?: if (ignoreMissingMembers) {
+                        continue
+                    } else {
+                        throw CodegenException("Protocol test defines data for member shape `${key.value}`, but member shape was not found on structure shape ${shape.id}")
```

Comment:
Hmm no, because the `Instantiator` doesn't have access to it. But the test's ID is logged before invoking the instantiator, so there should be no issue debugging if this happens.